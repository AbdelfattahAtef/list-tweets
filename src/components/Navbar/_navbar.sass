@import "../../styles/_variables.sass"
@import "../../styles/_breakpoints.sass"
@import url('https://fonts.googleapis.com/css?family=ZCOOL+QingKe+HuangYou')

.navbar
  padding: 10px
  position: fixed
  width: calc(100% - 20px)
  top: 0
  background: $color--white
  border-bottom: 3px solid $color--light-blue
  color: $color--light-blue
  z-index: 99
  display: none
  i, button
    cursor: pointer
    outline: 0
    border: 0
    background-color: $color--white
    &:focus
      outline: 0
  +breakpoint(xsmax)
    display: block
