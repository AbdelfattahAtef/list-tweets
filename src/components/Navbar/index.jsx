import React, {Component} from "react";
import "./_navbar.sass";

class Navbar extends Component {
    state = {
        toggleSidebar: false
    };

    toggleSidebarDisplay = () => {
        this.setState(prevState => ({
            toggleSidebar: !prevState.toggleSidebar
        }), () => (
            this.props.toggleSidebar(this.state.toggleSidebar)
        ));

    };
    render(){
        return (
            <div className="navbar">
                {/*<i className="fa fa-bars" onClick={this.toggleSidebarDisplay}/>*/}
                <button onClick={this.toggleSidebarDisplay}>
                    {this.state.toggleSidebar ? 'Hide Sidebar' : 'Open Sidebar'}
                </button>
            </div>
        );
    }
}

export default Navbar;
