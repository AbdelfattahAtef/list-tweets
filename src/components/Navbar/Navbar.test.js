import React from 'react';
import Navbar from '../Navbar';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<Navbar />', () => {
    it('should render Navbar component', () => {
        const wrapper = shallow(<Navbar />);
        expect(wrapper.find('.fa-bars').length).toBe(1);
        expect(wrapper.find('.fa-close').length).toBe(0);
        wrapper.setState({toggleSidebar: true});
        expect(wrapper.find('.fa-bars').length).toBe(0);
        expect(wrapper.find('.fa-close').length).toBe(1);
    });
    it('matches the snapshot', () => {
        const tree = shallow(<Navbar />);
        expect(toJson(tree)).toMatchSnapshot();
    })
});
