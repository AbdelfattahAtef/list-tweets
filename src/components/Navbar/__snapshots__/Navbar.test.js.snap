// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> matches the snapshot 1`] = `
<div
  className="navbar"
>
  <button
    onClick={[Function]}
  >
    <i
      className="fa fa-bars"
    />
  </button>
</div>
`;
